### PR TITLE
Implement Repair method on client

### PR DESCRIPTION
**Pull request by zaibon**

### CODE REVIEW

File: client/check.go [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+			case <-ctx.Done():
+				// in case we already know something is corrupted, we don't need
+				// to check other blokcs
+				return
+
+			default:
+				checks, err := store.ObjectsCheck(ids)
+				if err != nil {
+					log.Errorf("error getting check status on shard %s: %v", shard, err)
+					cErr <- err
```
Comment (iwanbk) [inline client/check.go]: When checking for #240, i was sure that there is another code that has same bug.
Now i remember, this code.

In this case, if there are more than one error, only one goroutine could exit.
Subsequent goroutine will be blocked by the channel

File: client/check.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+				}
+
+				for _, check := range checks {
+					if check.Status != CheckStatusOk {
+						// signal we found a corrupted or missing block
+						cStatus <- check.Status
```
Comment (iwanbk) [inline client/check.go]: same as bug with cErr above

File: client/check_test.go [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+	meta, err := c.Write(key, data, []string{})
+	require.NoError(t, err, "fail write data")
+
+	// Check status is ok after a write
+	status, err := c.Check(meta.Key)
+	require.NoError(t, err, "fail to check object")
+	assert.Equal(t, CheckStatusOk, status)
+
+	// corrupt file by removing a block
```
Comment (iwanbk) [inline client/check_test.go]: need test for removing more than one block, make sure all goroutines could exit

File: client/client_test.go [hunk 4]
```diff
@@ -429,6 +429,57 @@ func BenchmarkWriteFilesSizes(b *testing.B) {
 	}
 }
 
+func TestIssue225(t *testing.T) {
```
Comment (iwanbk) [inline client/client_test.go]: Is it mean #225 has been fixed? What is the actual problem?

File: client/repair.go [hunk 5]
```diff
@@ -0,0 +1,183 @@
...
+	ErrAllReplicateCorrupted = fmt.Errorf("All the replicate are corrupted, repair impossible")
+	// ErrRepairSupport is returned when a block is not using replication or distribution
+	ErrRepairSupport = fmt.Errorf("block is not using replication or distribution, repair impossible")
+)
+
+// Repair repairs a broken file.
+// If the file is distributed and the ammout of corrupted chunks is acceptable,
+// we recreate the missing chunks.
+// Id the file is replicated and we still have one valid replicate, we create the missing replicate
+// till we reach the replication number configured in the policy
```
Comment (iwanbk) [inline client/repair.go]: > till we reach the replication number configured in the policy

why not fixes all shards?

File: client/client_test.go (hunk 4 above)
Comment (zaibon) [reply client/client_test.go]: That was a test to try to reproduce the #255 when we were still thinking that downloading multiple time the same file corrupted files. This test prove it doesn't.

File: client/repair.go (hunk 5 above)
Comment (zaibon) [reply client/repair.go]: you want to replicate randomly again, not using the same shards.

**Pull request merged**